browser_history/generic.py

```diff
@@ -377,3 +377,22 @@ def default(self, o):
             )
 
         return json_string
+
+    def save(self, filename, output_format="infer"):
+        """
+        Saves history or bookmarks to a file. Infers the type from the given filename
+        extension. If the type could not be inferred, it defaults to csv.
+
+        :param filename: the name of the file.
+        :param output_format: (optional)One the formats in `csv`, `json`, `jsonl`.
+            If not given, it will automatically be inferd from the file's extension
+        """
+        if output_format == "infer":
+            file_extension = os.path.splitext(filename)[1][1:]
+            if file_extension == "json":
+                output_format = "json"
+            else:
+                output_format = "csv"
```

Comment:
Instead of hardcoding these, you could use the `_format_map` to get these automatically.
(You have missed jsonl here which wouldn't happen if you use `_format_map`)

Reply:
I reckon `_format_map` could lead to problems if we ever decide to implement a format that has a file extension different from its name (I'm not sure if these exist tho). I could implement some kind of extension map if hardcoding values like this is too much of a bad practice.

Reply:
I don't know of any formats that have names different from their extension. Even if they do exist, we could perhaps use the extension as the name.
An extension map seems like a good idea, but unnecessary as of right now since we have only 3 formats.